bacon/model/Transaction.swift

```diff
@@ -42,11 +58,105 @@ struct Transaction: Codable, Equatable {
             throw InitializationError(message: "`amount` must be greater than 0")
         }
 
-        self.date = date
-        self.type = type
-        self.frequency = frequency
-        self.category = category
-        self.amount = amount
-        self.description = description
+        _date = date
+        _type = type
+        _frequency = frequency
+        _category = category
+        _amount = amount
+        _description = description
+    }
+
+    /// Notifies all observers of changes to self.
+    /// This should be called after any mutation to a Transaction instance.
+    private func notifyObserversOfSelf() {
+        log.info("Notifying observers of new self")
+        notifyObservers(self)
+    }
+
+    // Public computed properties
+    var date: Date {
+        get {
+            return _date
+        }
+
+        set(newDate) {
+            log.info("Setting date=\(newDate)")
+            _date = newDate
+            notifyObserversOfSelf()
+        }
+    }
+
+    var type: TransactionType {
+        get {
+            return _type
+        }
+
+        set(newType) {
+            log.info("Setting type=\(newType)")
+            _type = newType
+            notifyObserversOfSelf()
+        }
+    }
+
+    var frequency: TransactionFrequency {
+        get {
+            return _frequency
+        }
+
+        set(newFrequency) {
+            log.info("Setting frequency=\(newFrequency)")
+            _frequency = newFrequency
+            notifyObserversOfSelf()
+        }
+    }
+
+    var category: TransactionCategory {
+        get {
+            return _category
+        }
+
+        set(newCategory) {
+            log.info("Setting category=\(newCategory)")
+            _category = newCategory
+            notifyObserversOfSelf()
+        }
     }
+
+    var amount: Decimal {
+        get {
+            return _amount
+        }
+
+        set(newAmount) {
+            log.info("Setting amount=\(newAmount)")
+            _amount = newAmount
+            notifyObserversOfSelf()
+        }
+    }
+
+    var description: String {
+        get {
+            return _description
+        }
+
+        set(newDescription) {
+            log.info("Setting description=\(newDescription)")
+            _description = newDescription
+            notifyObserversOfSelf()
+        }
+    }
+
+}
+
+extension Transaction: Equatable {
+
+    static func == (lhs: Transaction, rhs: Transaction) -> Bool {
```

Comment:
wouldn't this defeat the identification purpose whereby 2 transactions of same value should be different transactions?

Reply:
Not really, there is the triple-equal `===` to test for identity. I guess this is useful for checking if 2 separate transactions have the same "contents", i.e. 2 different transactions that are equivalent.

Granted, I can't think of a situation where this is useful, but I thought I should leave the Equatable conformance in in case it is utilized in another part of the code.

Reply:
If the user's new value is equal to old value when he tries to edit, we can use == to check if there is even a need to notify observers. But even if we notify anyway it should be okay as well, but this can be a possible use case

Reply:
Resolved using didSet()